### PR TITLE
fix: [PL-40551]: reduce project and organizations api call default page size to 20

### PR DESCRIPTION
**Pull request by namanharness**
With current pageSize limit there is high latency on below api’s when high number of concurrent users are accessing the application.
- `/ng/api/aggregate/projects`
- `/ng/api/aggregate/organizations`

Changes: - 
- Changed the pageSize of mentioned API Calls to 20 across application

Tested in new and old nav scenarios

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
- MessageMetadata: `trigger messagecheck`
- GitLeaks: `trigger gitLeaks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

### CODE REVIEW

**Review by chetanvekariya-harness — DISMISSED** [review]

**Comment by namanharness** [comment]
trigger messagecheck

File: cypress/support/85-cv/verifyStep/constants.ts [hunk 1]
```diff
@@ -52,7 +52,7 @@ export const pipelinesSummaryFetchCall = `/pipeline/api/pipelines/summary/NG_Doc
 export const pipelinesYamlFetchCall = `/pipeline/api/yaml-schema?*`
 // git sync call
 export const gitSyncCall = `/ng/api/git-sync?routingId=${accountId}&accountIdentifier=${accountId}&orgIdentifier=${org}&projectIdentifier=${project}`
-export const aggregateProjectsCall = `/ng/api/aggregate/projects?routingId=${accountId}&accountIdentifier=${accountId}&pageIndex=0&pageSize=50&*`
+export const aggregateProjectsCall = `/ng/api/aggregate/projects?routingId=${accountId}&accountIdentifier=${accountId}&pageIndex=0&pageSize=20&*`
```
Comment (mayankVermaHarness) [inline cypress/support/85-cv/verifyStep/constants.ts:55]: use constant `DEFAULT_PAGE_SIZE_OPTION`

File: src/modules/45-projects-orgs/utils/utils.ts [hunk 2]
```diff
@@ -194,7 +194,7 @@ export const projectsPageQueryParamOptions = {
     return {
       ...params,
       page: params.page ?? 0,
-      size: params.size ?? COMMON_DEFAULT_PAGE_SIZE
+      size: params.size ?? COMMON_PAGE_SIZE_OPTIONS[0]
```
Comment (mayankVermaHarness) [inline src/modules/45-projects-orgs/utils/utils.ts:197]: This value is 25. Expected?

**Review by mayankVermaHarness — DISMISSED** [review]

File: cypress/support/85-cv/verifyStep/constants.ts (hunk 1 above)
Comment (namanharness) [reply cypress/support/85-cv/verifyStep/constants.ts:55]: imports are not allowed in this file

File: src/modules/10-common/hooks/useDefaultPaginationProps.ts [hunk 3]
```diff
@@ -35,7 +35,7 @@ export const useDefaultPaginationProps = (props: PaginationProps): PaginationPro
         position: PopoverPosition.TOP
       }
     },
-    pageSizeOptions: PL_NEW_PAGE_SIZE ? COMMON_PAGE_SIZE_OPTIONS : [10, 20, 50, 100],
+    pageSizeOptions: PL_NEW_PAGE_SIZE ? COMMON_PAGE_SIZE_OPTIONS : DEFAULT_PAGE_SIZE_OPTIONS,
```
Comment (abhinavrastogi-harness) [inline src/modules/10-common/hooks/useDefaultPaginationProps.ts:38]: `PL_NEW_PAGE_SIZE` flag was added 8 months ago. can we just remove it entirely and have a much simpler flow with the correct defaults?

Comment (namanharness) [reply src/modules/10-common/hooks/useDefaultPaginationProps.ts:38]: do you mean we should remove this FF from harness-core-ui?

Comment (mayankVermaHarness) [reply src/modules/10-common/hooks/useDefaultPaginationProps.ts:38]: Let's check if `PL_NEW_PAGE_SIZE` is GA'ed yet, and then take up clean-up of FF separately

**Review by mayankVermaHarness — APPROVED** [review]

**Review by PratyushGar — APPROVED** [review]

**Pull request merged**